tests/framework/microvm.py

```diff
@@ -114,6 +119,10 @@ def kill(self):
                 shell=True
             )
 
+        if self._memory_events_queue and not self._memory_events_queue.empty():
```

Comment:
I would rather have this check in the Microvm destructor so that this code gets implicitly called and survives future refactorings.

Reply:
Pytest doesn't care about exceptions thrown in the destructor. By the time the `microvm` is no longer referenced, the test has already ended (successfully).
Here's how it looks:

```
...Exception ignored in: <function Microvm.__del__ at 0x7faa47c5ae18>
Traceback (most recent call last):
  File "/home/ANT.AMAZON.COM/aghecen/work/firecracker/github/firecracker/tests/framework/microvm.py", line 114, in __del__
    self._memory_events_queue.get())
host_tools.memory.MemoryUsageExceededException: Memory usage exceeded maximum threshold. Usage: 2484 MiB.
```

Reply:
That s****. Ok then, we can leave it as it is.